src/kernel.hpp

```diff
@@ -350,6 +346,16 @@ struct cvk_kernel_argument_values {
             if (resource)
                 resource->release();
         }
+        std::lock_guard<std::mutex> lock(m_lock);
+        if (--m_descriptor_sets_refcount == 0) {
+            m_is_enqueued = false;
```

Comment:
This variable (and the associated getters) should really have been named: `m_was_used_by_in_flight_command`. We were never resetting it. With your change `m_is_used_by_in_flight_command` would become more appropriate. To be clear I'm not requesting you make those changes.